### PR TITLE
Address a possible (albeit unlikely) out-of-bounds read/write for ove…

### PR DESCRIPTION
**Pull request by Castaglia**
…rly long async response lines.

Thanks to Bruno Henc for reporting this issue.

### CODE REVIEW

File: src/response.c [hunk 1]
```diff
@@ -342,25 +342,40 @@ void pr_response_send_async(const char *resp_numeric, const char *fmt, ...) {
...
   va_end(msg);
 
-  buf[sizeof(buf) - 1] = '\0';
+  buf[sizeof(buf)-1] = '\0';
 
   resp_last_response_code = pstrdup(resp_pool, resp_numeric);
   resp_last_response_msg = pstrdup(resp_pool, buf + len + 1);
 
-  sstrcat(buf + res, "\r\n", max_len - res);
+  if (res < 0) {
```
Comment (Castaglia) [inline src/response.c:366]: The cause of the issue here was that we were not checking the `pr_vsprintf` return value to see whether it would have filled the buffer, and thus proceeded blithely on.  Now, we take such conditions into account.

File: tests/api/response.c [hunk 2]
```diff
@@ -354,6 +355,60 @@ START_TEST (response_send_async_test) {
 }
 END_TEST
 
+START_TEST (response_send_async_too_large_test) {
+  char buf[PR_TUNABLE_BUFFER_SIZE * 2], expected[PR_TUNABLE_BUFFER_SIZE];
```
Comment (Castaglia) [inline tests/api/response.c:359]: This input buffer size is guaranteed to exceed that of the `pr_response_send_async()` function deliberately, so that we test its boundary condition handling.
